ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsLocator.java

```diff
@@ -119,9 +119,15 @@
     // Depending on whether this is a key-based lookup, determine which metadata method to use.
     // If we don't have keys, find the metadata for all partitions since we'll run the query for
     // all partitions of the state store rather than a particular one.
-    final List<PartitionMetadata> metadata = keys.isEmpty() || isRangeScan
-        ? getMetadataForAllPartitions(filterPartitions, keySet)
-        : getMetadataForKeys(keys, filterPartitions);
+    //For issue #7174. Temporarily turn off metadata finding for a partition with keys
+    //if there are more than one key.
+    final List<PartitionMetadata> metadata;
+    if (keys.size() == 1 && keys.get(0).getKey().size() == 1 && !isRangeScan) {
```

Comment:
I thought the issue happens if we have a multi-column key, not if they're searching for multiple instances of a key.  If `keys.size() > 1`, this is like `KEY_COL IN (1, 2, 3, 4, 5);`, which I would think shouldn't trigger the bug.

You can always look at the schema or even grab one of the keys to see if there's more than one column to determine if it's multi-column.

This change would technically still work, I'm just trying not to entirely get rid of the precise forwarding optimization where we can still keep it.